src/database/giveaway.rs

```diff
@@ -0,0 +1,177 @@
+use diesel::prelude::*;
+
+use chrono::NaiveDateTime;
+
+use crate::{models::*, Message};
+
+use super::Database;
+
+impl Database {
+    pub async fn start_giveaway<S: Into<String>>(
+        &self,
+        message: &Message,
+        end_time: NaiveDateTime,
+        max_winners: i64,
+        prize: S,
+    ) -> Result<i64, anyhow::Error> {
+        use crate::schema::{Giveaways as GiveawaysSchema, Giveaways::dsl::Giveaways};
+        let giveaway = NewGiveaway {
+            message_id: message.id.0,
+            channel_id: message.channel_id.0,
+            end_time,
+            max_winners,
+            prize: prize.into(),
+        };
```

Comment:
Should there be a way to create a giveaway without starting it?

Reply:
I don't think this is worth implementing as it would add unnecessary complexity to the start command or require a new command, and in most cases the user simply wants to immediately start a giveaway.

Reply:
I think we'll have to consult the end-users (masterminds) about this.
@Esinko @ahnl @developerfromjokela

Reply:
Also in case they do want it, we can always make an issue of it since this commit is already quite large and feature ful.

Reply:
I'll mark this as resolved, if the feature is deemed necessary it can be implemented in another PR